src/main/java/uk/gov/register/util/HashValue.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.register.util;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonValue;
+import uk.gov.register.core.HashingAlgorithm;
+import uk.gov.register.exceptions.HashDecodeException;
+
+public class HashValue {
+    private final String value;
+    private final String hashingAlgorithm;
+
+    public HashValue(HashingAlgorithm hashingAlgorithm, String value) {
+        this.hashingAlgorithm = hashingAlgorithm.toString();
+        this.value = value;
+    }
+
+    public HashValue(String hashingAlgorithm, String value) {
+        this.hashingAlgorithm = hashingAlgorithm;
+        this.value = value;
+    }
+
+    @JsonValue
+    public String encode() {
+        return hashingAlgorithm + ":" + value;
+    }
+
+    @JsonIgnore
+    public String getValue() {
+        return value;
+    }
+
+    public static HashValue decode(HashingAlgorithm hashingAlgorithm, String encodedHash) {
+        if (!encodedHash.startsWith(hashingAlgorithm.toString())) {
+            throw new HashDecodeException(String.format("Hash \"%s\" has not been encoded with hashing algorithm \"%s\"", hashingAlgorithm, encodedHash));
+        }
+
+        String[] parts = encodedHash.split(hashingAlgorithm + ":");
+
+        if (parts.length != 2) {
+            throw new HashDecodeException(String.format("Cannot decode hash %s", encodedHash));
+        }
+
+        return new HashValue(hashingAlgorithm, parts[1]);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || o.getClass() != this.getClass()) return false;
+
+        HashValue that = (HashValue) o;
+
+        return this.encode().equals(that.encode());
```

Comment:
`equals` should not be based on a method as the method could be changed by someone who doesn't notice that `equals` is using it. It needs to be based on the `value` and `hashingAlgorithm` properties themselves. See other examples in `Item.equals` and `Entry.equals`